15-components-lab_v16/src/02-numberguessing.jsx

```diff
@@ -1,17 +1,40 @@
-/**
- * Lab 02:
- *
- * Implement a number guessing game component.
- *
- * The component should provide an input field for the user to guess a
- * number picked randomly by the computer (from 1 to 1,000).
- * When typing a number, the component should print "too low" or "too high" 
- * as a hint to the user.
- *
- * To make the game interesting, computer should sometimes cheat and print wrong messages.
- */
-
-import React from 'react';
-
-export default React.createClass({
-});
+import React from 'react';
+
+
+class GuessNumber extends React.Component {
+    
+   
+    constructor(props) {
+        super(props);
+        this.state = { randomNumber : Math.round(Math.random() * 1000) , message : "" };
+      }
+
+
+      check = (e) =>{
+        let guessedNo = parseInt(document.getElementById("guessedNo").value)

```

Comment:
Don't use IDs nor getElementById in React. You never know how your components are going to be used.

Reply:
Even querySelector is not recommended, as it's hard to limit its effect only to your rendered data.

In future lessons we'll learn how to query DOM elements from React. Until then use `value` and `onChange` to connect the input element to a value in state.